protocol/src/main/java/org/fido/iot/protocol/To2ServerService.java

```diff
@@ -165,6 +165,16 @@ protected void doProveDevice(Composite request, Composite reply) {
       payload.set(Const.THIRD_KEY, nonce7);
       payload.set(Const.FOURTH_KEY, replacementKey);
 
+      // TO2SetupDevicePayload is signed by Owner2, which in this code is this owner
+      // probably replacing the manufacturer-owner.
+      try (CloseableKey key =
+          new CloseableKey(getStorage().getOwnerSigningKey(ownerPublicKey))) {
+        payload = getCryptoService().sign(
+            key.get(), payload.toBytes(), getCryptoService().getCoseAlgorithm(ownerPublicKey));
```

Comment:
Should be signed by the Owner2 key that is stored in variable 'replacementKey'.

Reply:
Fixed by f9f0de6.